dowhy/causal_graph.py

```diff
@@ -79,13 +79,14 @@ def __init__(self,
         # Adding node attributes
         self._graph = self.add_node_attributes(observed_node_names)
 
-    def view_graph(self, layout="dot", size=(8, 6), file_name="causal_model"):
-        out_filename = "{}.png".format(file_name)
+    def view_graph(self, layout="dot", size=(8, 6), file_name=None):
+        if file_name is None:
+            file_name = tempfile.mkstemp(suffix='.png', prefix='causal_graph_')
         try:
             import pygraphviz as pgv
             agraph = nx.drawing.nx_agraph.to_agraph(self._graph)
             agraph.graph_attr.update(size="{},{}!".format(size[0], size[0]))
-            agraph.draw(out_filename, format="png", prog=layout)
+            agraph.draw(file_name, format="png", prog=layout)
```

Comment:
What if the user provides a different extension, "f.jpg"? In this case, the `format` argument should be changed here. 
Does it make sense to add code to automatically infer the extension by string parsing, and then use that as `format`?

Reply:
The code being patched does not allow format changes either. There are several patterns possible here:
- document that function creates PNG figures (and if the user wants to call the output file `"plot.xls"`, that their choice).
- allow a parametrization of the output format. Inferring the format from the extension name can be convenient. I also often like to be able to specify a format. For example, this could be a named parameter `format` that infers the format from the extension in the file name if `None`. One can argue that this would allow one to write something like `view('foo.png', format='jpg') but I like separating the specification of a file format from the file name.

Reply:
Got it, thanks @lgautier That's okay to merge. 
Do you mind signing the CLA agreement for this contribution? You will see it as a post above. 

Also, the PR is marked as a draft PR, so it cannot be merged. Can you change the status to "Ready for review/not a draft
"?